plugins/modules/identity/ipa/ipa_sudorule.py

```diff
@@ -237,7 +237,7 @@ def sudorule_add_allow_command(self, name, item):
         return self._post_json(method='sudorule_add_allow_command', name=name, item={'sudocmd': item})
 
     def sudorule_add_allow_command_group(self, name, item):
-        return self._post_json(method='sudorule_add_allow_command_group', name=name, item={'sudocmdgroup': item})
+        return self._post_json(method='sudorule_add_allow_command', name=name, item={'sudocmdgroup': item})
```

Comment:
I'm not sure calling a method when its function name is another method is good, what do you think about also changing `sudorule_add_allow_command_group` function name to `sudorule_add_allow_command` now that it's using `sudorule_add_allow_command` method?

Reply:
@TizeN85 I think it's good to fix this prior to [3.3.0 release](https://github.com/ansible-collections/community.general/issues/582#issuecomment-856764385) before end of June.
Feel free to ping me if you didn't have time to fix this, so I could do that.

Reply:
Hi @aminvakil, in fact there already is a function named sudorule_add_allow_command at line 234 :)
Maybe we should try to do like with sudorule_remove_host, sudorule_remove_host_host and sudorule_remove_host_hostgroup functions?
Something like this? (NOT TESTED YET) :
```
          def sudorule_remove_host(self, name, item):
              return self._post_json(method='sudorule_remove_host', name=name, item=item)
  
          def sudorule_remove_host_host(self, name, item):
              return self.sudorule_remove_host(name=name, item={'host': item})
  
          def sudorule_remove_host_hostgroup(self, name, item):
              return self.sudorule_remove_host(name=name, item={'hostgroup': item})
  
          def sudorule_add_allow_cmd(self, name, item):
            return self._post_json(method='sudorule_add_allow_command', name=name, item=item)
  
          def sudorule_add_allow_command(self, name, item):
            return self.sudorule_add_allow_cmd(name=name, item={'sudocmd': item})

        def sudorule_add_allow_command_group(self, name, item):
            return self.sudorule_add_allow_cmd(name=name, item={'sudocmdgroup': item})
```

Reply:
I didn't know there was a function with the same name, maybe we should keep it just the way it is right now.

Also extending `add_allow_cmd` function which takes `sudocmd` and `sudocmdgroup` as arguments instead of changing functions to call it is better IMO. Different parts of code should change with this change too, so now that it's getting more and more complicated, I'm not really sure about this change anymore as I don't know if there is that much value in this :)

Reply:
Yes, maybe this should be part of a later bigger PR :)

So, should I "resolve" this conversation? :)

Reply:
@TizeN85 Sure, go ahead!
Also I'll be happy to review if you wanted to open a PR whenever you wanted to about this ;)